time_tracker_api/users/users_namespace.py

```diff
@@ -25,17 +25,30 @@
             description='Email of the user that belongs to the tenant',
             example=faker.email(),
         ),
-        'role': fields.String(
+        'role': NullableString(
             title="User's Role",
             max_length=50,
             description='Role assigned to the user by the tenant',
-            example=faker.word(['admin']),
+            example=faker.word(['time-tracker-admin']),
         ),
     },
 )
 
 user_response_fields.update(common_fields)
 
+user_input_fields = ns.model(
+    'UserInput',
+    {
+        'role': NullableString(
```

Comment:
I am not a restplus expert, but this looks a little bit odd ( this field is **Nullable** and  **required** at the same time)
```
 'role': NullableString(
            title="User's Role",
            required=True,
            max_length=50,
            description='Role assigned to the user by the tenant',
            example=faker.word(['time-tracker-admin']),
        ),
```
Would not it better if we make this field optional??

Reply:
we'll use this field to allow PUT calls to switch user's roles:
with input :
{'role': 'time-tracker-admin'} we give the role

with input : 
{'role': null} we remove the role

for that reason is nullable & string, and it is required because this is the field we want to change for this endpoint. 

Does that make sense?

Reply:
Yes, it does. 
However, since the URI for this endpoint for this resource is `/users/<user-id>/roles` you are going to add roles ( POST operation) and remove roles ( DELETE operation), I would suggest using the POST and DELETE operations instead of PUT. 
It would be different if the URI were `/users/<user-id>`, in that case I would use the PUT operation.

Reply:
@juanultimate  We used PUT because looking at our code base, this requirement it looked pretty similar to an update operation(In our codebase all updates are managed via PUT).

I understand your point, probably from the perspective of a UI dev, it is weird to use PUT in something that has no entity. Maybe a POST would seem more natural. About the DELETE operation, not that sure. We may need to discuss this in a meeting.